### PR TITLE
refactor: [M3-7329] - MUI Migration - `SRC > Components > Crumbs`

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- MUI migration for the `Crumbs` component
- I really want to get this epic done so we can update to React 18 and drop @mui/styles

## Changes  🔄
- Runs codemod on `packages/manager/src/components/Breadcrumb/Crumbs.tsx`
- Find and replaces instances of `import { useTheme } from '@mui/styles';` to `import { useTheme } from '@mui/material';` because we intend to remove `@mui/styles` at the end of this migration
- Rewrites `Breadcrumb` Storybook story because the old one was broken

## Preview 📷

> [!note]
> No UI changes are expected

| before | after |
|--------|-------|
|     Screenshot 2023-10-25 at 3 14 26 PM   |   Screenshot 2023-10-25 at 3 14 32 PM    |

### Verification steps 
- Checkout this PR
- View various Breadcrumbs throughout Cloud Manager
- Verify Breadcrumbs look identical to what is at https://cloud.linode.com
- Optional: view the Storybook story

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

**Review by cpathipa — APPROVED** [review]
LGTM! verified Breadcrumbs and no regressions found.

File: packages/manager/src/components/Breadcrumb/Breadcrumb.stories.tsx
Comment (mjac0bs) [inline packages/manager/src/components/Breadcrumb/Breadcrumb.stories.tsx:1]: Story looks good. I did notice that this component has its own little [`README.md`](https://github.com/linode/manager/blob/develop/packages/manager/src/components/Breadcrumb/README.md). Does it make more sense to document relevant and useful info from that file in this Storybook story and ditch the markdown? I'm pretty sure that's the only component I've ever seen with its own README.

File: packages/manager/src/components/Breadcrumb/Crumbs.tsx [hunk 1]
```diff
@@ -115,11 +112,10 @@ export const Crumbs = React.memo((props: Props) => {
           </div>
         );
       })}
-
       {/*
-        for prepending some SVG or other element before the final crumb.
-        See users detail page for example
-      */}
+      for prepending some SVG or other element before the final crumb.
+      See users detail page for example
```
Comment (mjac0bs) [inline packages/manager/src/components/Breadcrumb/Crumbs.tsx:117]: A bit out of scope for this PR: but is this comment actually true for the user details page? I'm not seeing anything being prepended before the final crumb.

**Review by mjac0bs — APPROVED** [review]
Crumbs looked good throughout Cloud! 🍪 

Left a couple of minor, non-blocking things.

**Pull request merged**